angular/lib/scripts/verify-module.mjs

```diff
@@ -0,0 +1,50 @@
+import {glob} from 'glob';
+import path from 'path';
+import {fileURLToPath} from 'url';
+import {readFile} from 'fs/promises';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const pathRegExp = /\\/g;
+function normalizePath(str) {
+	return str.replace(pathRegExp, '/');
+}
+
+const COMPONENTS_FOLDER = path.join(__dirname, '../src/components');
+
+const files = (
+	await glob('**/*.component.ts', {
+		cwd: COMPONENTS_FOLDER,
+		nodir: true,
+	})
+).map((file) => normalizePath(path.join(COMPONENTS_FOLDER, file)));
+
+const stuffToExportRegex = /(@Directive|@Component)\({[\S\s]+?}\)\s*export class ([a-zA-Z]*)/g;
```

Comment:
We have a use case of "empty" `@Directives()` in the headless, but I think your regex is more than enough for components (in general you have at least the `selector`).